optimum/intel/openvino/modeling.py

```diff
@@ -468,13 +468,30 @@ def _from_transformers(
         )
 
         config.save_pretrained(save_dir_path)
-        return cls._from_pretrained(
+        model = cls._from_pretrained(
             model_id=save_dir_path,
             config=config,
             load_in_8bit=load_in_8bit,
             quantization_config=quantization_config,
             **kwargs,
         )
+        if {"token_embeddings", "sentence_embedding"}.issubset(
+            {name for output in model.model.outputs for name in output.names}
+        ):  # Sentence Transormers outputs
+            warnings.warn(
+                "This model is Sentence Tranfromers converted model. Please use OVModelForSentenceTransformer explicitly for this model."
+            )
+            from .modeling_sentence_transformers import OVModelForSentenceTransformer
+
+            model = OVModelForSentenceTransformer(
+                model=model.model,
+                config=config,
+                model_save_dir=save_dir_path,
+                quantization_config=quantization_config,
+                model_id=model_id,
+                **kwargs,
+            )
```

Comment:
we should test this part as well

Reply:
It was mistake in the pipeline. This logic was moved to _from_pretrained, test was added.